stan/math/prim/scal/prob/exp_mod_normal_rng.hpp

```diff
@@ -3,38 +3,70 @@
 
 #include <stan/math/prim/scal/err/check_consistent_sizes.hpp>
 #include <stan/math/prim/scal/err/check_finite.hpp>
-#include <stan/math/prim/scal/err/check_not_nan.hpp>
 #include <stan/math/prim/scal/err/check_positive_finite.hpp>
-#include <stan/math/prim/scal/fun/constants.hpp>
-#include <stan/math/prim/scal/meta/length.hpp>
+#include <stan/math/prim/scal/meta/max_size.hpp>
 #include <stan/math/prim/scal/meta/VectorBuilder.hpp>
-#include <stan/math/prim/scal/prob/normal_rng.hpp>
 #include <stan/math/prim/scal/prob/exponential_rng.hpp>
-#include <stan/math/prim/scal/meta/include_summand.hpp>
-#include <stan/math/prim/scal/fun/value_of.hpp>
+#include <stan/math/prim/scal/prob/normal_rng.hpp>
 #include <boost/random/normal_distribution.hpp>
 #include <boost/random/variate_generator.hpp>
 #include <string>
 
 namespace stan {
   namespace math {
-
-    template <class RNG>
-    inline double
-    exp_mod_normal_rng(double mu,
-                       double sigma,
-                       double lambda,
+    /**
+     * Return a pseudorandom Exponentially modified normal variate for the
+     * given location, scale, and inverse scale using the specified random
+     * number generator.
+     *
+     * mu, sigma, and lambda can each be either scalars or vectors. All vector
+     * inputs must be the same length.
+     *
+     * @tparam T_loc Type of location parameter
+     * @tparam T_scale Type of scale parameter
+     * @tparam T_inv_scale Type of inverse scale parameter
+     * @tparam RNG type of random number generator
+     * @param mu (Sequence of) location parameter(s)
+     * @param sigma (Sequence of) scale parameter(s)
+     * @param lambda (Sequence of) inverse scale parameter(s)
+     * @param rng random number generator
+     * @return Exponentially modified normal random variate
+     * @throw std::domain_error if mu is infinite, sigma is nonpositive,
+     * or lambda is nonpositive
+     * @throw std::invalid_argument if vector arguments are not the same length
```

Comment:
So I think this should say "container arguments" where you have "vector arguments".